### ISSUE TITLE
Questions  About Reward Model

### ISSUE THREAD

**Issue by hashword0428** [OpenRLHF/OpenRLHF#542]
Hi thx so much about this brilliant project which benefits me a lot!! But I still have some questions about the Reward Model, I would be appreciated if you could provide me some illustration!!

**Question 1:  Why doesn't the  Reward Model (OpenRLHF/Llama-3-8b-rm-mixture) have a value head?**
I tried to load the reward model directly:
```python
model_name_or_path = "OpenRLHF/Llama-3-8b-rm-mixture"
model = AutoModel.from_pretrained(model_name_or_path, torch_dtype=torch.bfloat16 )
```
and get the model structure as follows:

```
LlamaModel(
  (embed_tokens): Embedding(128256, 4096)
  (layers): ModuleList(
    (0-31): 32 x LlamaDecoderLayer(
      (self_attn): LlamaSdpaAttention(
        (q_proj): Linear(in_features=4096, out_features=4096, bias=False)
        (k_proj): Linear(in_features=4096, out_features=1024, bias=False)
        (v_proj): Linear(in_features=4096, out_features=1024, bias=False)
        (o_proj): Linear(in_features=4096, out_features=4096, bias=False)
        (rotary_emb): LlamaRotaryEmbedding()
      )
      (mlp): LlamaMLP(
        (gate_proj): Linear(in_features=4096, out_features=14336, bias=False)
        (up_proj): Linear(in_features=4096, out_features=14336, bias=False)
        (down_proj): Linear(in_features=14336, out_features=4096, bias=False)
        (act_fn): SiLU()
      )
      (input_layernorm): LlamaRMSNorm((4096,), eps=1e-05)
      (post_attention_layernorm): LlamaRMSNorm((4096,), eps=1e-05)
    )
  )
  (norm): LlamaRMSNorm((4096,), eps=1e-05)
  (rotary_emb): LlamaRotaryEmbedding()
)
```
I expected to get something like `(v_head): Linear(in_features=..., out_features=1... bias=False)` just as I found in other opensource reward model like internlm/internlm2-1_8b-reward etc., could you give some hints about this question?

**Question2:  As for reward model, do we initialize the value head every time no matter whether we are training it or just using it?**

I found that in `train_ppo.py`, we get reward model by `_get_reward_model`.
But when I dived into this function, I found in the `__init__`that:

```python
# https://github.com/OpenRLHF/OpenRLHF/blob/main/openrlhf/models/model.py#L155
setattr(self, value_head_prefix, nn.Linear(config.hidden_size, 1, bias=False))
```
By using `setattr`, it seems that we would use `nn.Linear(config.hidden_size, 1, bias=False)` to replace  the reward model's value_head no matter the reward model(OpenRLHF/Llama-3-8b-rm-mixture) has a value_head or not. But in PPO training we just want to use the reward model directly without refining its structure.  I'm really confused about this and don't know if I make something wrong here.

**Comment by hijkzzz** [comment]
OpenRLHF/Llama-3-8b-rm-mixture has a value_head, just print the state_dict

**Comment by hashword0428** [comment]
> OpenRLHF/Llama-3-8b-rm-mixture has a value_head, just print the state_dict

oh yes,  i just read the source codes of Transformer `.from_pretrained` method, and I know where the probs are:
- First, I should not use something like `AutoModel` to load RM directly, I should use cls_class from `_get_reward_model`, that solves my Question 1.
- Second, As for  `_get_reward_model`, it always initialize a value_head first, and then when we use `.from_pretrained` to load model, if the model itself has value_head, it would replace the initialized value_head, otherwise we just use the value_head we initilaize, thus for separating RM inference and Training, that solves my Question 2.

Thx for your patience and reply!
